server/utils/nunjucksSetup.ts

```diff
@@ -340,6 +345,44 @@ export function registerNunjucks(app?: express.Express): Environment {
     return word[0].toUpperCase() + word.substr(1).toLowerCase()
   })
 
+  njkEnv.addFilter('createOffenderLink', (licence: FoundProbationRecord): string => {
+    const isTimedOutLicence = licence?.licenceStatus === 'TIMED_OUT'
+    const isHardStopLicence = licence?.kind === 'HARD_STOP'
+
+    if (isTimedOutLicence && licence.versionOf) {
+      return `/licence/create/id/${licence.licenceId}/licence-changes-not-approved-in-time`
+    }
+
+    if (
+      (isTimedOutLicence && !isHardStopLicence) ||
+      (isHardStopLicence && licence.licenceStatus === LicenceStatus.IN_PROGRESS)
+    ) {
+      return `/licence/create/nomisId/${licence.nomisId}/prison-will-create-this-licence`
+    }
+
+    if (isHardStopLicence) {
+      return `/licence/create/id/${licence.licenceId}/licence-created-by-prison`
+    }
+
+    if (!licence.licenceId) {
+      return `/licence/create/nomisId/${licence.nomisId}/confirm`
+    }
+
+    return `/licence/create/id/${licence.licenceId}/check-your-answers`
+  })
+
+  njkEnv.addFilter('getlicenceStatusForSearchResults', (licence: FoundProbationRecord): LicenceStatus => {
+    if (licence.isReviewNeeded) {
+      return LicenceStatus.REVIEW_NEEDED
+    }
+    const isHardStopLicence = licence?.kind === 'HARD_STOP'
+    return isHardStopLicence ? LicenceStatus.TIMED_OUT : <LicenceStatus>licence.licenceStatus
+  })
+
+  njkEnv.addFilter('getreleaseDateForSearchResults', (releaseDate: string): string => {
```

Comment:
typo but could we not do this in the controller?

Reply:
we apparantly call this `DateShort` (e.g: `jsonDtToDateShort`) so `cvlDateAsDateShort` ?

But better in controller if possible 

(we should be parsing dates as close to receiving them as possible - ideally in the client but definitely not when rendering)

Reply:
looking through the code and I can see why you've done it that way, maybe just call it `cvlDateToDateShort` for now

(we need to readdress this at some point!)

Reply:
If we handle `'not found'` logic in the backend then we can use `cvlDateAsDateShort`.
Ya sure, will make it `cvlDateToDateShort` for now.